caas/kubernetes/provider/k8s.go

```diff
@@ -154,6 +154,51 @@ func (k *kubernetesClient) deleteNamespace() error {
 	return errors.Trace(err)
 }
 
+func (k *kubernetesClient) EnsureSecret(appName string, containerSpec *caas.ContainerSpec) error {
```

Comment:
comment
also, pass in ImageDetails, not container

Reply:
I use containerSpec.Name to keep the secret name unique, as a pod may have multiple containers with different images used within (but the same app name).

Reply:
missed that. i think maybe pass in the secret name (constructed outside the method) and image details.
keeps the method a bit tighter in terms of responsibility